ui/packages/consul-ui/app/components/topology-metrics/card/index.hbs

```diff
@@ -6,10 +6,11 @@
 </a>
 {{else}}
 <a class="topology-metrics-card"
+  data-test-topology-metrics-card
   href={{if
    (and (env 'CONSUL_NSPACES_ENABLED') (not-eq @item.Namespace @service.Namespace))
-    (href-to "dc.services.show.index" @item.Datacenter @item.Name params=(hash nspace=@item.Namespace))
-    (href-to "dc.services.show.index" @item.Name)
+    (href-to this.hrefPath @item.Datacenter @item.Name params=(hash nspace=@item.Namespace))
+    (href-to this.hrefPath @item.Name)
   }}
   data-permission={{service/intention-permissions @item}}
   id="{{@item.Namespace}}{{@item.Name}}"
```

Comment:
Not related to this PR, but I noticed that a super edgy-case thing here:

Given a namespace like "johns"
And a namespace like "john"
And a service like "service"
And a service like "ervice"
Then id would be the same for both

```
johns/ervice
id="johnservice"
john/service
id="johnservice"
```

Bit of an edge case, but it wouldn't hurt here to use some sort of separator or encoding I think (or if possible, not use unique id's at all) Obvs for another PR if you wanted to do something here

Reply:
yes, I agree. This should be refactored. I took a stab and this change would cascade into multiple files getting touched. Noted for another PR in Asana.